### ISSUE TITLE
should ngx-translate service subscribe calls be unsubscribed to avoid mem leaks?

### ISSUE THREAD

**Issue by andreimcristof** [ngx-translate/core#796]
Hi! The docs said that any questions should be addressed here, so I have a question please:

In the usage guide it says the translations done with the service should be subscriptions i.e. 

> With the service, it looks like this:
> ```
> translate.get('HELLO', {value: 'world'}).subscribe((res: string) => {
>     console.log(res);
>     //=> 'hello world'
> })```

My question is, should I actually also UNsubscribe in my onDestroy event, **for each** of these resources I subscribe to, whenever I translate from service directly ? The Usage guide has no mention of this aspect or I could not find it. I ask because potentially these could be a lot of subscriptions to unsub from, and it feels redundant because you might have some form of cleaning already built-in, so I just want to be sure what I should do. Thanks!

**Comment by RoyiNamir** [comment]
@andreimcristof I also thought about this and docs doesnt say nothing about it. I think we should 
 use the `takeWhile` approach.....

**Comment by jlberrocal** [comment]
Maybe the main reason for not mention anything in docs is because this is an Angular pattern which is not directly related to this library, as you mention you could unsubscribe for avoid mem leaks, however you should also not be using the get method too much, please try to use always the directive or pipe approach

**Comment by manandkumaar** [comment]
Its not required to call unsubscribe as the translate.get() method will automatically complete the subject.

Below is the source code for your reference.

![image](https://user-images.githubusercontent.com/2328960/42050696-467ddd86-7b27-11e8-8aed-5e3d46695b7e.png)

**Comment by enoh-barbu** [comment]
only for methods like stream you should unsubscribe!

**Comment by riccardomessineo** [comment]
@manandkumaar 
This should definitely be written in the docs!

**Comment by enoh-barbu** [comment]
@riccardomessineo I thought it's obviously you don't need to do that, if you know js

**Comment by appleseedexm** [comment]
This should be in the docs. The lack of need for an unsubscribe is a pattern that is not rxjs specific, but specific for the likes of angular and apparently ngx-translate. 'If you know js' (read as rxjs) you should always assume you'd need to unsubscribe.

**Comment by enoh-barbu** [comment]
dude, how can you unsubscribe from something that you've never been subscribed too? are you for real?

**Comment by jagadeesh93** [comment]
> only for methods like `stream` you should unsubscribe!

How to unsubscribe the translate.stream??

**Comment by suprishi** [comment]
What about `onLangChange`? Do we need to unsubscribe from that?

**Comment by Sampath-Lokuge** [comment]
@suprishi Yes, Please see here: https://github.com/ngx-translate/core/issues/319#issuecomment-259935416

**Comment by Sampath-Lokuge** [comment]
@jagadeesh93 It is just a normal unsubscribe way. e.g. https://github.com/ngx-translate/core/issues/319#issuecomment-259935416

**Comment by virgiliofrn** [comment]
> cribe the tr

subscript = new Subscription();
 const sub= this.translate......
 ....).subscribe();
        this.subscript.add(sub);


 ngOnDestroy() {
    this.subscript.unsubscribe();
  }
